python/astro_metadata_translator/translators/decam.py

```diff
@@ -297,3 +298,57 @@ def fix_header(cls, header, instrument, obsid, filename=None):
                       log_label, header["FILTER"], obstype)
 
         return modified
+
+    @classmethod
+    def read_all_headers(cls, filename, primary=None):
+        """Read all relevant headers from the given file.
+
+        Returns all non-guide headers and does not include the primary
+        header.
+
+        Parameters
+        ----------
+        filename : `str`
+            Path to a file in a format understood by this translator.
+        primary : `dict`-like, optional
+            The primary header obtained by the caller. This is sometimes
+            already known, for example if a system is trying to bootstrap
+            without already knowing what data is in the file. For many
+            instruments where the primary header is the only relevant
+            header, the primary header will be returned with no further
+            action.
+
+        Yields
+        ------
+        headers : iterator of `dict`-like
+            Each relevant header in turn.
+
+        Notes
+        -----
+        DECam data use INHERIT=T so all returned headers will be merged
+        with the primary header.
+        """
+        # Circular dependency so must defer import
+        from ..headers import merge_headers
+
+        # Since we want to scan many HDUs we use astropy directly to keep
+        # the file open rather than continually opening and closing it
+        # as we go to each HDU.
+        with fits.open(filename) as fits_file:
+            # Astropy does not automatically handle the INHERIT=T in
+            # DECam headers so the primary header must be merged.
+            first_pass = True
+
+            for hdu in fits_file:
+                if first_pass:
+                    if not primary:
+                        primary = hdu.header
+                    first_pass = False
+                    continue
+
+                header = hdu.header
+                if "CCDNUM" not in header:
```

Comment:
I assume it's because the primary header doesn't have a CCDNUM and it will skip it. Maybe there are other HDUs that should be skipped as well. I didn't write the code when it was in obs_decam. Maybe it was @parejkoj ?

Reply:
Can you please point me to the obs_decam lines you're referring to? I need some context here (if it even was me).

Reply:
Sorry. The SHA1 won't work from here.  Try this: https://github.com/lsst/obs_decam/commit/1f9b698deb4976870652e993550d77935ea47b67

Reply:
and I've just realized you were asking about the CCDNUM test and not the CCDNUM usage the line below. Ignore me. The answer is the primary header does not have a CCDNUM and I just wanted to be sure I wasn't reading a non-detector HDU (I don't know enough about decam to know if there are any other headers).